src/fairseq2/recipes/lm/_online_finetune/_common.py

```diff
@@ -255,47 +272,63 @@ def copy_state(src_module: nn.Module, tgt_module: nn.Module):
         tgt_param = tgt_state[name_edited]
         tgt_param.data.copy_(src_param.data.to(tgt_param.device))
 
+
 def sync_weights_with_vllm(train_model, vllm_model, trainer_process_group):
     """
     trainer_process_group must connect training process with vllm_model processes
     """
     for name, p in train_model.module.named_parameters():
         name = name.replace("._checkpoint_wrapped_module", "")
         # print(f'sync call {name}')
-        handle = vllm_model.collective_rpc.remote("update_weight",
-                                        args=(name, p.dtype, p.shape))
+        handle = vllm_model.collective_rpc.remote(
+            "update_weight", args=(name, p.dtype, p.shape)
+        )
         trainer_process_group.broadcast(p, src=0, stream=torch.cuda.current_stream())
         ray.get(handle)
 
+
 def find_first_value(lst, value):
     return next((i for i, x in enumerate(lst) if x == value), None)
 
-def generate_rollouts(prompts: List[str], dp_gang: Gang, vllm_model, sampling_params=None):
-    prompts_to_generate = [None]*dp_gang.size
+
+def generate_rollouts(
+    prompts: List[str],
+    dp_gang,
+    vllm_model,
+    operation: str = "rollout",
+    sampling_params=None,
+):
+    if operation not in ["rollout", "reward"]:
+        raise ValueError("Operation must be either 'rollout' or 'reward'.")
+
+    prompts_to_generate = [None] * dp_gang.size
     if dp_gang.rank == 0:
         dp_gang.gather_object(prompts, prompts_to_generate, 0)
     else:
         dp_gang.gather_object(prompts, None, 0)
-
     if dp_gang.rank == 0:
         rank_batch_sizes = [len(l) for l in prompts_to_generate]
         flat_request_list = []
         for rank_prompts in prompts_to_generate:
             flat_request_list.extend(rank_prompts)
-        
-        rollouts = vllm_model.rollout_from_model(flat_request_list, sampling_params=sampling_params)
 
-        rollouts_to_scatter = []
-        rollouts_per_rank = [None]
+        if operation == "rollout":
+            outputs = vllm_model.rollout_from_model(
+                flat_request_list, sampling_params=sampling_params
+            )
+
+        elif operation == "reward":
+            outputs = vllm_model.reward_from_model(flat_request_list)
```

Comment:
I wanted to keep both of these within the same `generate_rollouts()`, but not sure if this is the best way to do it.